events/handler.go

```diff
@@ -273,7 +313,19 @@ func (self *events) AddEvent(e *Event) error {
 	self.updateEventList(e)
 	watchesToSend := self.findValidWatchers(e)
 	for _, watchObject := range watchesToSend {
-		watchObject.channel <- e
+		watchObject.eventChannel.GetChannel() <- e
 	}
 	return nil
 }
+
+func (self *events) CleanupWatcher() {
+	for i := range self.deadWatchAlert {
+		_, ok := self.watchers[i]
+		if !ok {
+			glog.Infof("Could not find watcher instance %v", i)
+		}
+		self.watcherLock.RLock()
```

Comment:
since CleanupWatcher is called in a goroutine and the channel deadWatchAlert does not terminate, I thought by making a defer call to Unlock() would mean it would never release the watcherLock